src/cloud-api-adaptor/pkg/userdata/provision.go

```diff
@@ -212,18 +311,25 @@ func ProvisionFiles(cfg *Config) error {
 	ctx, cancel := context.WithTimeout(bg, duration)
 	defer cancel()
 
-	provider, err := newProvider(ctx)
-	if err != nil {
-		return fmt.Errorf("failed to create UserData provider: %w", err)
-	}
+	// some providers provision config files via process-user-data
+	// some providers rely on cloud-init provisin config files
+	// all providers need calculate the hash value for attesters usage
```

Comment:
Can we make the digest calculation its own subcommand, instead of choosing a code path?

Reply:
I'm thinking we'll remove `cloud-init` incrementally, the process-user-data need add other providers implementation step by step. And the hash calculation should be just one step of the whole process. So I prefer keep it in same command, wdyt?

Reply:
we have to see how the final picture looks like, but we can add a dedicated command later, if required